cdap-formats/src/main/java/io/cdap/cdap/format/io/JsonStructuredRecordDatumReader.java

```diff
@@ -155,6 +172,27 @@ protected Object decodeUnion(Decoder decoder, Schema unionSchema) throws IOExcep
                                         token));
   }
 
+  @Override
+  protected Object decode(Decoder decoder, Schema schema) throws IOException {
+    Schema nonNullableSchema = schema.isNullable() ? schema.getNonNullable() : schema;
+    Schema.LogicalType logicalType = nonNullableSchema.getLogicalType();
+    if (logicalTypeAsString && logicalType != null) {
+      switch (logicalType) {
+        case DATE:
+          return LocalDate.parse(decoder.readString(), DateTimeFormatter.ISO_DATE);
+        case TIME_MILLIS:
+        case TIME_MICROS:
```

Comment:
quick question, will precision loose if we handle MILLIS the same with MICROS? Or it does not matter when we convert them to String?

Reply:
yea it will take care of it upto nano seconds